TileStache/Goodies/VecTiles/transform.py

```diff
@@ -646,22 +646,67 @@ def calculate_default_place_scalerank(shape, properties, fid, zoom):
     return shape, properties, fid
 
 
-# create a new layer from the boundaries of polygons in the
-# base layer, subtracting any sections of the boundary which
-# intersect other polygons.
-#
-# the purpose of this is to provide us a shoreline / river
-# bank layer from the water layer without having any of the
-# shoreline / river bank draw over the top of any of the base
-# polygons.
-#
-# properties on the lines returned are the same as the
-# polygon feature they came from.
-#
-# any features in feature_layers[layer] which aren't
-# polygons will be ignored.
+def _make_new_properties(props, props_instructions):
+    """
+    make new properties from existing properties and a
+    dict of instructions.
+
+    the algorithm is:
+      - where a key appears with value True, it will be
+        copied from the existing properties.
+      - where it's a dict, the values will be looked up
+        in that dict.
+      - otherwise the value will be used directly.
+    """
+    new_props = dict()
+
+    for k, v in props_instructions.iteritems():
+        if v is True:
+            # this works even when props[k] = None
+            if k in props:
+                new_props[k] = props[k]
```

Comment:
`val = props.get(k)` gives None if `props[k] = None`. I can't remember why I thought that was important at the time... perhaps I was considering using None instead of True for the copy...?

I guess I don't have strong feelings either way.